src/controller/java/templates/partials/decode_value.zapt

```diff
@@ -51,9 +51,17 @@ if ({{source}}.IsNull()) {
     );
   {{else}}
     {{#if_chip_enum type}}
-      std::string {{target}}ClassName = "{{asJniClassName type null cluster}}";
-      std::string {{target}}CtorSignature = "({{asJniSignature type null cluster false}})V";
-      chip::JniReferences::GetInstance().CreateBoxedObject<{{asUnderlyingBasicType type}}>({{target}}ClassName.c_str(), {{target}}CtorSignature.c_str(), static_cast<{{asUnderlyingBasicType type}}>({{source}}), {{target}});
+        std::string {{target}}ClassName = "{{asJniClassName type null cluster}}";
+        std::string {{target}}CtorSignature = "({{asJniSignature type null cluster false}})V";
+      {{~#if (isStrEqual (asUpperCamelCase type) (asUpperCamelCase "vendor_id"))~}}
+        chip::JniReferences::GetInstance().CreateBoxedObject<{{asUnderlyingBasicType type}}>({{target}}ClassName.c_str(), {{target}}CtorSignature.c_str(), static_cast<{{asUnderlyingBasicType type}}>({{source}}), {{target}});
+      {{else}}
+        {{#if_is_atomic type}}
```

Comment:
The "is atomic means it's an untyped enumN/bitmapN type" thing is extremely unclear.

We should have "is weakly typed enum" and "is weakly typed bitmap" helpers instead or something.